data/src/main/java/com/catalincretu/data/feature/local/model/NewsLocalSource.java

```diff
@@ -0,0 +1,65 @@
+package com.catalincretu.data.feature.local.model;
+
+import com.catalincretu.data.feature.local.mapper.ArticleEntityToArticleMapper;
+import com.catalincretu.data.feature.model.Article;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+
+public class NewsLocalSource {
+    private final ArticlesDao dao;
+
+    public NewsLocalSource(ArticlesDao dao) {
+        this.dao = dao;
+    }
+
+    public Single<List<Article>> getArticles() {
+        return dao.queryArticles().map(new ArticleEntityToArticleMapper());
+    }
+
+    public Completable saveArticles(List<Article> articles) {
+        List<ArticleEntity> entityList = new ArrayList<>();
+
+        //convert an Article to an ArticleEntity
+        for (Article article : articles) {
+            ArticleEntity articleEntity = new ArticleEntity(
+                    article.imageUrl,
+                    article.url,
+                    article.content,
+                    article.description,
+                    article.title,
+                    ""
+            );
+            entityList.add(articleEntity);
+        }
+        return (Completable) dao.insertArticles(entityList)
+                .observeOn(Schedulers.io())
+                .subscribe();
+    }
+
+   /* public Single<ArticleEntity> getArticleItem(int id) {
+        return dao.queryArticleItem(id);
+    }
```

Comment:
let's cleanup the commented code, no need for commented code, we can use git for this hisotry